### PR TITLE
Update references to woocommerce_payments_server

### LINKED ISSUE #9823: Update references to server docker containers
Related to https://github.com/Automattic/transact-platform-server/issues/6717

We've recently updated the names of server docker containers. We need to update references of old server container names in WooPayments.

### PR DESCRIPTION
**Pull request by achyuthajoy**
Fixes #9823

#### Changes proposed in this Pull Request

* Updates server container prefix from `woocommerce_payments_server` to `transact_platform_server`
* Updates server repo folder name from `wcp-server` to `transact-platform-server`
* Updates the GitHub secret `WCP_SERVER_REPO` to `TRANSACT_PLATFORM_SERVER_REPO`
* Updates the env variable `WCP_SERVER_BRANCH` to `TRANSACT_PLATFORM_SERVER_BRANCH`
* Updates docs where required

#### Testing instructions

* Follow the instructions on https://github.com/Automattic/woocommerce-payments/tree/update/server-container-name/tests/e2e to set up the local e2e environment and confirm that the environment is working as expected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9824` or branch name `update/server-container-name` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/server-container-name) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 93fb9910e7d5d85d35c5002801f36f04b2525057
- Build time: 2024-11-28 04:13:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -10,22 +10,22 @@ on:
...
-  E2E_SLACK_TOKEN:       ${{ secrets.E2E_SLACK_TOKEN }}
-  E2E_USE_LOCAL_SERVER:  false
-  E2E_RESULT_FILEPATH:   'tests/e2e/results.json'
-  WC_MIN_SUPPORTED_VERSION: '7.6.0'
-  NODE_ENV: 'test'
-  FORCE_E2E_DEPS_SETUP: true
+  E2E_GH_TOKEN:                  ${{ secrets.E2E_GH_TOKEN }}
+  WCP_DEV_TOOLS_REPO:            ${{ secrets.WCP_DEV_TOOLS_REPO }}
+  WCP_DEV_TOOLS_BRANCH:          'trunk'
+  TRANSACT_PLATFORM_SERVER_REPO: ${{ secrets.TRANSACT_PLATFORM_SERVER_REPO }}
```
Comment (htdat) [inline .github/workflows/e2e-test.yml:16]: Without updating this secret, I think it's still working until we reuse the old server repo name. 

But anyway, we would still update the secret to use the new one now.

**Review by htdat — APPROVED** [review]
Overall, LGTM! 

After merging this PR, we might need to cherry-pick its commit for the `trunk` branch too. E2E tests on the server are using the `trunk` branch.

File: .github/workflows/e2e-test.yml (hunk 1 above)
Comment (achyuthajoy) [reply .github/workflows/e2e-test.yml:16]: @htdat I have already created a secret `TRANSACT_PLATFORM_SERVER_REPO` with the new repo url

**Review by alopezari — APPROVED** [review]
LGTM!

**Pull request merged**